### DISCUSSION TITLE
Tinycam RTSP Audio Issue

### DISCUSSION THREAD

**Discussion by revenant-81** [gtxaspec/wz_mini_hacks#93] [Q&A]
Hello,

I've setup RTSP and have no issues streaming video and audio through VLC. When attempting to enable audio, it throws an error stating the codec isn't supported.

"Unsupported codec. Switch camera audio codec to Linear PCM, G.711 A-law (PCMA), AAC."

Is there a way to configure the audio codec RTSP uses? The Wyze Official RTSP was able to stream audio through TinyCam.

Many thanks.

**Comment by gtxaspec** [comment]
this ses to be a tinycam bug.  it would be worthwhile to report it to their developers.

the settings for the rtsp audio stream are `s16le           PCM signed 16-bit little-endian`, it's a commom audio format.

we tap into the audio stream before, the camera re-encodes it, sending it to the apps or web rtc.

**Reply by mrlt8** [reply]
Actually, I believe s16le isn't a supported a supported codec for RTSP. That's why we had to re-encode it to aac in the bridge.

**Reply by gtxaspec** [reply]
interesting, v4l2rtspserver supports it though, so it kind of works for us then

**Reply by mrlt8** [reply]
https://datatracker.ietf.org/doc/html/rfc3551#section-4.5.11

> .. transmitted in network byte order (most significant byte first).

**Reply by clee604** [reply]
Hi @mrlt8, are you using rtsp-simple-server to re-encode the audio stream? I use it currently to proxy my RTSP streams and was wondering if you could share your ffmpeg command line to re-encode to AAC? I got it working with a simple `ffmpeg -rtsp_transport udp -i rtsp://localhost:8554/wyze -c:v copy -c:a aac -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH` but I'm seeing "max delay reached, need to consume packet" errors in the log.

**Reply by mrlt8** [reply]
We're actually re-encoding the audio before rtsp-simple-server. 

Have you tried specifying the audio codec and bitrate?

This is what we're using in the bridge:
https://github.com/mrlt8/docker-wyze-bridge/blob/b0b41dbe1987fa60c7cd7c3f7f0e2e6ef9c42352/app/wyze_bridge.py#L593

**Reply by clee604** [reply]
Thanks, I'm setting AAC as the codec but didn't set an explicit output bitrate. I'm not really knowledgeable in python but are you explictly setting an output bitrate in your code? I see a reference to audio rate but that looks like its on the input side?
